applications/StructuralMechanicsApplication/custom_elements/updated_lagrangian.h

```diff
@@ -0,0 +1,336 @@
+///@{
+///@}
+///@name Type Definitions
+///@{
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Updated Lagrangian element for 2D and 3D geometries.
+
+/**
+ * Implements a total Lagrangian definition for structural analysis.
+ * This works for arbitrary geometries in 2D and 3D
+ */
+
+class UpdatedLagrangian
+    : public BaseSolidElement
+{
+public:
+    ///@name Type Definitions
+    ///@{
+    ///Reference type definition for constitutive laws
+    typedef ConstitutiveLaw ConstitutiveLawType;
+    ///Pointer type for constitutive laws
+    typedef ConstitutiveLawType::Pointer ConstitutiveLawPointerType;
+    ///Type definition for integration methods
+    typedef GeometryData::IntegrationMethod IntegrationMethod;
+
+    /// Counted pointer of UpdatedLagrangian
+    KRATOS_CLASS_POINTER_DEFINITION(UpdatedLagrangian);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    UpdatedLagrangian(IndexType NewId, GeometryType::Pointer pGeometry);
+    UpdatedLagrangian(IndexType NewId, GeometryType::Pointer pGeometry, PropertiesType::Pointer pProperties);
+
+    /// Destructor.
+    ~UpdatedLagrangian() override;
+
+    ///@}
+    ///@name Operators
+    ///@{
+    ///@}
+    ///@name Operations
+    ///@{
+    
+    /**
+     * Called to initialize the element.
+     * Must be called before any calculation is done
+     */
+    void Initialize() override;
+    
+    /**
+     * Returns the currently selected integration method
+     * @return current integration method selected
+     */
+    //TODO: ADD THE OTHER CREATE FUNCTION
+    Element::Pointer Create(IndexType NewId, NodesArrayType const& ThisNodes, PropertiesType::Pointer pProperties) const override;
+
+    /**
+     * This function provides the place to perform checks on the completeness of the input.
+     * It is designed to be called only once (or anyway, not often) typically at the beginning
+     * of the calculations, so to verify that nothing is missing from the input
+     * or that no common error is found.
+     * @param rCurrentProcessInfo
+     */
+    int Check(const ProcessInfo& rCurrentProcessInfo) override;
+
+    //std::string Info() const;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    //      virtual String Info() const;
+
+    /// Print information about this object.
+    //      virtual void PrintInfo(std::ostream& rOStream) const;
+
+    /// Print object's data.
+    //      virtual void PrintData(std::ostream& rOStream) const;
+    ///@}
+    ///@name Friends
+    ///@{
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+    
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    std::vector<double> mDetF0; // Historical total elastic deformation measure determinant
```

Comment:
the Element has a  method

SetValueOnIntegrationPoints which you can use for that